instrumentation/opentelemetry_beam/src/opentelemetry_vm_statistics.erl

```diff
@@ -0,0 +1,185 @@
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+setup() ->
+    setup(?current_meter).
+
+setup(Meter) ->
+    Instruments =
+        [otel_meter:create_observable_counter(
+           Meter, 'erlang.vm.statistics.output',
+           #{description => <<"Total number of bytes output to ports.">>,
+             unit => <<"By">>}),
+         otel_meter:create_observable_counter(
+           Meter, 'erlang.vm.statistics.received',
+           #{description => <<"Total number of bytes received through ports.">>,
+             unit => <<"By">>}),
+         otel_meter:create_observable_counter(
+           Meter, 'erlang.vm.statistics.context_switches',
+           #{description => <<"Total number of context switches "
+                              "since the system started.">>}),
+         otel_meter:create_observable_gauge(
+           Meter, 'erlang.vm.statistics.dirty_cpu_run_queue_length',
+           #{description => <<"Length of the dirty CPU run-queue.">>}),
+         otel_meter:create_observable_gauge(
+           Meter, 'erlang.vm.statistics.dirty_io_run_queue_length',
+           #{description => <<"Length of the dirty IO run-queue.">>}),
+         otel_meter:create_observable_counter(
+           Meter, 'erlang.vm.statistics.reductions',
+           #{description => <<"Total reductions.">>}),
+         otel_meter:create_observable_gauge(
+           Meter, 'erlang.vm.statistics.run_queues_length',
+           #{description => <<"Length of normal run-queues.">>}),
+         otel_meter:create_observable_counter(
+           Meter, 'erlang.vm.statistics.runtime',
+           #{description => <<"The sum of the runtime for all threads "
+                              "in the Erlang runtime system. "
+                              "Can be greater than wall clock time.">>,
+             unit => <<"s">>}),
+         otel_meter:create_observable_counter(
+           Meter, 'erlang.vm.statistics.wallclock_time',
+           #{description => <<"Information about wall clock. "
+                              "Same as erlang_vm_statistics_runtime "
+                              "except that real time is measured.">>,
+             unit => <<"s">>}),
+         otel_meter:create_observable_counter(
+           Meter, 'erlang.vm.statistics.garbage_collection_number_of_gcs',
+           #{description => <<"Garbage collection: number of GCs.">>}),
+         otel_meter:create_observable_counter(
+           Meter, 'erlang.vm.statistics.garbage_collection_bytes_reclaimed',
+           #{description => <<"Garbage collection: bytes reclaimed.">>,
+             unit => <<"By">>}),
+         otel_meter:create_observable_counter(
+           Meter, 'erlang.vm.statistics.garbage_collection_words_reclaimed',
+           #{description => <<"Garbage collection: words reclaimed.">>})],
+
+    otel_meter:register_callback(Meter, Instruments, fun vm_stats/1, []),
```

Comment:
Ooops, completely missed these were created without callbacks, haha.

So yea, this is good.

But still I'd suggest grouping them so they aren't all read even if you only want certain of them? What do you think?